p2p/base_reactor.go

```diff
@@ -28,6 +28,9 @@ type Reactor interface {
 	//
 	// CONTRACT: msgBytes are not nil.
 	Receive(chID byte, peer Peer, msgBytes []byte)
+
+	// InitAddPeer is called by switch before peer is started.
+	InitAddPeer(peer Peer) Peer
```

Comment:
I like `InitPeer` better

Reply:
updated @melekes @ebuchman @xla  would you like review again?